# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/dietlibc/pkg_patch/pkg_nasm.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- nasm-0.98.36/nasmlib.c~	Thu Feb  6 02:44:09 2003
+++ nasm-0.98.36/nasmlib.c	Sun Sep  7 23:23:29 2003
@@ -372,10 +372,8 @@
     if (posn >= r->stepsize * LAYERSIZ(r))
 	return 0;		/* Return 0 for undefined entries */
     while (r->layers > 0) {
-	ldiv_t l;
-	l = ldiv (posn, r->stepsize);
-	r = r->u.b.data[l.quot];
-	posn = l.rem;
+	r = r->u.b.data[posn/r->stepsize];
+	posn = posn%r->stepsize;
 	if (!r)
 	    return 0;		/* Return 0 for undefined entries */
     }
@@ -408,14 +406,12 @@
     result = r;
 
     while (r->layers > 0) {
-	ldiv_t l;
 	struct RAA **s;
-	l = ldiv (posn, r->stepsize);
-	s = &r->u.b.data[l.quot];
+	s = &r->u.b.data[posn / r->stepsize];
 	if (!*s)
 	    *s = real_raa_init (r->layers - 1);
 	r = *s;
-	posn = l.rem;
+	posn = posn % r->stepsize;
     }
 
     r->u.l.data[posn] = value;
